src/daemon.py

```diff
@@ -41,22 +40,40 @@ def main(self, sleep_time: int) -> None:
         while True:
             time.sleep(sleep_time)
             end_block = web_3.eth.block_number
-            fetched_hashes = self.instance.get_settlement_hashes(start_block, end_block)
+            fetched_hashes = get_tx_hashes_by_block(web_3, start_block, end_block)
             all_hashes = fetched_hashes + unchecked_hashes
             unchecked_hashes = []
             while len(all_hashes) > 0:
                 single_hash = all_hashes.pop(0)
-                response_data = self.instance.get_solver_competition_data([single_hash])
-                if len(response_data) != 0:
-                    self.instance.get_order_surplus(response_data[0])
-                else:
+                if self.onchain_quasimodo_test(single_hash) or self.cow_endpoint_test(
```

Comment:
I find the convention used here a bit confusing. I would say that in case of success, the test should return TRUE while in case of failure, it should return FALSE.

Given that, then this line of code would look as follows:

`if (not self.onchain_quasimodo_test(single_hash)) or (not self.cow_endpoint_test(single_hash)):`